src/python/pants/engine/internals/parser_test.py

```diff
@@ -17,18 +20,38 @@ def test_imports_banned() -> None:
     assert "Import used in dir/BUILD at line 4" in str(exc.value)
 
 
-def test_unrecognized_symbol() -> None:
-    parser = Parser(
-        target_type_aliases=["tgt"],
-        object_aliases=BuildFileAliases(
-            objects={"obj": 0},
-            context_aware_object_factories={"caof": lambda parse_context: lambda _: None},
-        ),
-    )
-    prelude_symbols = BuildFilePreludeSymbols(FrozenDict({"prelude": 0}))
-    with pytest.raises(ParseError) as exc:
-        parser.parse("dir/BUILD", "fake", prelude_symbols)
-    assert (
-        str(exc.value)
-        == "Name 'fake' is not defined.\n\nAll registered symbols: ['caof', 'obj', 'prelude', 'tgt']"
-    )
+def test_unrecogonized_symbol() -> None:
+    def perform_test(extra_targets: list[str], dym: str) -> None:
+        parser = Parser(
+            target_type_aliases=["tgt", *extra_targets],
+            object_aliases=BuildFileAliases(
+                objects={"obj": 0},
+                context_aware_object_factories={"caof": lambda parse_context: lambda _: None},
+            ),
+        )
+        prelude_symbols = BuildFilePreludeSymbols(FrozenDict({"prelude": 0}))
+        with pytest.raises(ParseError) as exc:
+            parser.parse("dir/BUILD", "fake", prelude_symbols)
+        assert str(exc.value) == (
+            f"Name 'fake' is not defined.\n\n{dym}"
+            "If you expect to see more symbols activated in the below list,"
+            f" refer to {docs_url('enabling_backends')} for all available"
+            " backends to activate.\n\n"
+            "All registered symbols: ['caof', "
+            f"{str(extra_targets)[1:-1] + (', ') if len(extra_targets) != 0 else ''}"
+            "'obj', 'prelude', 'tgt']"
+        )
+
+    test_targs = ["fake1", "fake2", "fake3", "fake4", "fake5"]
+
+    dym_one = "Did you mean fake1?\n\n"
+    dym_two = "Did you mean fake2 or fake1?\n\n"
+    dym_many = "Did you mean fake5, fake4, or fake3?\n\n"
```

Comment:
Unlike the above being too complex, this is pretty simple and probably better to be inlined into the calls of `perform_test()`. You'll get better _proximity_, that the value is defined exactly where it's used.